### PR TITLE
Add custom Tumblr parameters

### PR DESCRIPTION
**Pull request by irace**

### CODE REVIEW

File: Example/XExtensionItem/ViewController.m [hunk 1]
```diff
@@ -35,12 +33,10 @@ - (void)action {
...
         mutableParameters.alternateContentRepresentations = @{ @"text/html": @"<p><strong>Apple’s website markup</strong></p>" };
-        
-        [mutableParameters addEntriesToUserInfo:({
-            TumblrCustomShareParameters *tumblrParameters = [[TumblrCustomShareParameters alloc] init];
-            tumblrParameters.customURLSlug = @"want-this-for-xmas";
-            tumblrParameters;
-        })];
+        mutableParameters.tumblrParameters = [[XExtensionItemTumblrParameters alloc] initWithCustomURLPathComponent:@"want-this-for-xmas"
+                                                                                                  requestedPostType:XExtensionItemTumblrPostTypeLink
+                                                                                                        consumerKey:@"a9fal494mglspj33lffaj4920vks"
```
Comment (mattbischoff) [inline Example/XExtensionItem/ViewController.m]: Maybe make it more obvious that this is a fake value so people don’t try to use it and then open issues when it doesn’t work.

File: Pod/Classes/Custom/Tumblr/XExtensionItemTumblrParameters.m [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    
+    if (self.consumerKey) {
+        [descriptionComponents addObject:[NSString stringWithFormat:@"consumerKey: %@", self.consumerKey]];
+    }
+    
+    if (self.correlationIdentifier) {
+        [descriptionComponents addObject:[NSString stringWithFormat:@"correlationIdentifier: %@", self.correlationIdentifier]];
+    }
+    
+    return [mutableDescription copy];
```
Comment (mattbischoff) [inline Pod/Classes/Custom/Tumblr/XExtensionItemTumblrParameters.m]: You’re not ever adding the stuff to the description.

Comment (irace) [inline Pod/Classes/Custom/Tumblr/XExtensionItemTumblrParameters.m]: :laughing:

**Comment by mattbischoff** [comment]
Good to merge once Travis likes you.

**Comment by irace** [comment]
I need to document `correlationIdentifier` and should probably add some tests as well, but thanks :+1:

**Pull request merged**